### ISSUE TITLE
Search block: When the button is inside, the input field has no visible focus indicator

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwentyone#908]
**Describe the bug**

When a search block is added and the button is placed inside, there is no visible focus indicator. 
-A blinking cursor is not enough to indicate focus.

The block editor sets the outline to none, so the themes focus outline is not visible.

```
.wp-block-search.wp-block-search__button-inside .wp-block-search__inside-wrapper .wp-block-search__input:focus {
    outline: none;
}
```

**To Reproduce**
1. Add a default search block. Duplicate it.
2. Select the button inside option
3. Duplicate the block and select the icon only, button inside options.
4. Go to the front.
5. focus on the blocks. See that there is a difference in the focus style for the 3 blocks.

**Expected behavior**

There needs to be a visible focus indicator.

**Screenshots**
![search-focus-indicator](https://user-images.githubusercontent.com/7422055/100747758-bfe4cb00-33e2-11eb-8de7-16c6d11cd8d5.gif)

**Editor version (please complete the following information):**
- WordPress version: 5.6-RC1-49715
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default?  Default.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Chrome, version 87.0.4280.66
